SingularityBase/src/main/java/com/hubspot/singularity/api/SingularityBounceRequest.java

```diff
@@ -5,23 +5,27 @@
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.common.base.Optional;
+import com.hubspot.singularity.SingularityShellCommand;
 import com.wordnik.swagger.annotations.ApiModelProperty;
 
 public class SingularityBounceRequest extends SingularityExpiringRequestParent {
 
   private final Optional<Boolean> incremental;
   private final Optional<Boolean> skipHealthchecks;
+  private final Optional<SingularityShellCommand> runBeforeKill;
 
   @JsonCreator
   public SingularityBounceRequest(@JsonProperty("incremental") Optional<Boolean> incremental, @JsonProperty("skipHealthchecks") Optional<Boolean> skipHealthchecks,
-      @JsonProperty("durationMillis") Optional<Long> durationMillis, @JsonProperty("actionId") Optional<String> actionId, @JsonProperty("message") Optional<String> message) {
+      @JsonProperty("durationMillis") Optional<Long> durationMillis, @JsonProperty("actionId") Optional<String> actionId, @JsonProperty("message") Optional<String> message,
+      @JsonProperty("runBeforeKill") Optional<SingularityShellCommand> runBeforeKill) {
```

Comment:
i think it'd be better to more explicit here, perhaps `runShellCommandBeforeKill`